### PR TITLE
add Schedule syntax

### LINKED ISSUE #44: Create environment agnostic versions of ZSchedule#*Env methods
Should be possible to do with patch. Something like:
```scala
  implicit class ZScheduleSyntax[R, A, B](sched: ZSchedule[R, A, B]) {
    def jittered0[R1 <: R with Clock](min: Double, max: Double)(implicit ev: R1 Mix Clock) =
      sched.jitteredEnv(min, max)(f => patch[R1, Clock].apply(c => new Clock { val clock = f(c.clock) }))
  }
```
/cc @iravid

### PR DESCRIPTION
**Pull request by mschuwalow**
resolves #44 

I also moved all syntax to a syntax object

/cc @ioleo

### CODE REVIEW

**Comment by mschuwalow** [comment]
@iravid pinging you as I need more help with naming here :)

File: core/shared/src/main/scala/zio/macros/delegate/ZScheduleSyntax.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import zio._
+import zio.clock.Clock
+import zio.random.Random
+import zio.duration.Duration
+
+trait ZScheduleSyntax {
+
+  implicit class ZScheduleSyntax[R, A, B](zSchedule: ZSchedule[R, A, B]) {
+
+    def delayed0[R1 <: R with Clock](f: Duration => Duration)(implicit ev: R Mix Clock): ZSchedule[R1, A, B] =
```
Comment (ioleo) [inline core/shared/src/main/scala/zio/macros/delegate/ZScheduleSyntax.scala]: I'm not sure about names with `0` suffix. I'd go with `_`.

**Comment by iravid** [comment]
`_` suffixes are (in Cats, ZIO and Haskell) used to denote a variant of a combinator that discards the results, usually resulting in an `F[Unit]` result rather than `F[A]`. This is probably not the case here.

**Comment by iravid** [comment]
... but unfortunately I don't have a better variant to suggest. I think that in this case too we should rename the `*Env` methods in ZIO and use those names here. The variants in ZIO are so painful and clunky that they would be very rarely used.

**Comment by mschuwalow** [comment]
> `_` suffixes are (in Cats, ZIO and Haskell) used to denote a variant of a combinator that discards the results, usually resulting in an `F[Unit]` result rather than `F[A]`. This is probably not the case here.

True, that breaks convention :(.

**Comment by mschuwalow** [comment]
> `_` suffixes are (in Cats, ZIO and Haskell) used to denote a variant of a combinator that discards the results, usually resulting in an `F[Unit]` result rather than `F[A]`. This is probably not the case here.

You are right, this creates the wrong intuition.


> ... but unfortunately I don't have a better variant to suggest. I think that in this case too we should rename the `*Env` methods in ZIO and use those names here. The variants in ZIO are so painful and clunky that they would be very rarely used.

We could do that.
What do you think about having names like jittered$ for our variants?

**Comment by iravid** [comment]
$?? Hehe that's a bit exotic for my tastes. What's your intuition for that? "jittered Apply"?

**Comment by mschuwalow** [comment]
> $?? Hehe that's a bit exotic for my tastes. What's your intuition for that? "jittered Apply"?

Not really, I was thinking that $ has no conventional meaning in scala as far as I'm aware and would lead to concise syntax here. Might be confusing for people that are aware of haskell though.

I'm really just bouncing ideas. If we can't come up with anything nice I propose we do the *Env naming you suggested

**Review by ioleo — APPROVED** [review]

**Comment by mschuwalow** [comment]
Let's get this merged for now and maybe iterate separately over naming

**Pull request merged**